### ISSUE TITLE
[RFC-0] Community organization

### ISSUE THREAD

**Issue by benchristel** [software-patterns/hci-language#1]
Tagging @bradfordboyle @ktchen14 @sannidhi @kalensk @amilkh since y'all have either asked me about this project in person or replied to my email. Everyone else I've contacted (including @gavmor, @kejadlen, @bitops, @jhamon, @professor, @pgoodwin), feel free to respond too! I want to work with all of you! I'm just shy about pushing my ideas too hard.

### What is this repo for?

**I want to discover better ways of creating software**, by creating it, reflecting on the process, and then, perhaps, writing a book about it. What exactly "better" means is up to all of us to figure out. I suspect we'll all approach the issue from slightly different angles, and that's great. One of the exciting features of our field is that there is no shortage of things that can be improved. :wink:

### How will this project work?

My vision is for this GitHub org to become a **semi-cooperative collective**, (kind of like Valve in structure; see their [new employee handbook](https://steamcdn-a.akamaihd.net/apps/valve/Valve_NewEmployeeHandbook.pdf)) where we collaboratively generate and discuss ideas. Any of us can then use those ideas (with attribution) in our own individually published works.

Here's how I imagine our creative process working:

1. We all write down our thoughts, feelings, and experiences related to software development. Anything and everything. Each item goes in its own file, to reduce the chance of merge conflicts. The writing does not have to be polished; it can be an outline, or even just a headline. We'll add our questions, comments, and answers inline in each others' files, so gradually the outlines will be filled in by discussions.
2. We can all contribute as much or as little as we want, as our schedules allow. 
3. Once our discussions begin to cool and solidify, we can begin the process of mining publishable materials from the collective work—though this doesn't mean we have to stop generating new ideas.
4. Members of the group can work solo on their own publishable projects, or form small groups—whatever we want. As we produce polished works, we should take care to give due credit to the originators of an idea—the git history will help us here. In this way, each of our individual projects will strengthen the reputation of others in the group.

I know this all probably sounds very abstract. If you'd like to see examples of the kind of granular idea-files I'd like us to start out with, I've already written [a handful of them](https://github.com/software-patterns/hci-language/tree/master/patterns) in this repo. (Note, though, that my examples are way too polished, due to my lack of prioritization ability.)

**Everything is open to change.** I am not committed to the "pattern language" format, the subject matter, or even the infrastructure. We don't have to use GitHub, or Git. We could make a wiki or a Google doc instead. But I think Git is nice so we can attribute work, as I described above.

### Next step: Roman vote

If, after reading all that, you're still interested in this project, please leave a thumbs-up/thumbs-down/meh reaction on this post to indicate your level of enthusiasm for my proposed creative process. If you're thumbs-down or meh, please also leave a comment to let me know what alternative organizing principles would make you excited to work on this project.

**Comment by ktchen14** [comment]
Hey Ben: I like the idea and the fact that you're trying a novel cooperative model in writing this book. However I'm not sure if I'll realistically have time to commit to this and don't want to say that I'll contribute and then not.

Is it cool if I just for the time being lurk the repo?

**Comment by benchristel** [comment]
@ktchen14 absolutely! You can hit the `Watch` button to subscribe to updates, if you like.

**Comment by professor** [comment]
I'm open to trying this format.

**Comment by amilkh** [comment]
Hmm, I think this is nice, I wonder if there are other people who have already done this?

Also, I tend to believe that in-person interactions are more valuable then written exchanges of information for creating something new, although this could be an interesting way to make a book.

Let's think of ideas on the bringing health / mindfulness into our software development.

Also, I think it would be great if we could base this work on XP and the Pivotal Labs way!

**Comment by benchristel** [comment]
@amilkh one similar project is the [C2 Wiki](http://wiki.c2.com/), started by Ward Cunningham. However, I believe it's frozen now.

**Comment by gavmor** [comment]
I'd be happy to introduce this to the Ex Pivots Slack...

**Comment by amilkh** [comment]
Cool! Where can we find the Ex Pivots Slack?

**Comment by benchristel** [comment]
@gavmor Please do!

> Cool! Where can we find the Ex Pivots Slack?

Yeah, are current pivots allowed?

**Comment by jhamon** [comment]
The ex-Pivot slack is https://pivotalalumni.slack.com and signup is through the google form at http://pivotal.fun . AFAIK it is for ex-pivots only. This thing has mostly just been passed from person to person so I'm sure some important folks have been overlooked. Can Berk Güder invited me after I left Pivotal.

This book project seems interesting but I have a different writing project right now that I'm focusing my creative energy on. I'm interested to contribute mainly as an editor / reader / feedbacker.

**Comment by pgoodwin** [comment]
I'm intrigued, but also over subscribed right now. It sure would be cool to be able to contribute in an ad hoc way like we did on Ward's (c2) Wiki.

**Comment by benchristel** [comment]
Hey @jhamon and @pgoodwin—no worries, I'm enthused to have you around in whatever capacity you have time for!

Phil, what sort of ad hoc mechanism were you hoping for? We can have wiki-style conversations in [the files](/software-patterns/workshop/tree/master/patterns) using [Markdown block quotes](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#blockquotes), e.g.

> [Alice] sup
> > [Bob] sup

source:

```markdown
> [Alice] sup
> > [Bob] sup
```

You all should have push access, so you don't have to fork this repo or anything. Just edit a file in the web UI and commit.

If that's too high-friction, there's always issues and comments. I'm happy to be the archivist of such things.

**Comment by kejadlen** [comment]
(Oops, mis-click!)
